### ISSUE TITLE
FexLinuxTests backlog

### ISSUE THREAD

**Issue by skmp** [FEX-Emu/FEX#1754]
An internal backlog of tests (mostly for me), so the tracker is not spammed with too many of these tickets. 

If you have any corner cases that you want tested, or ideas for edge case behaviour we don't cover feel free to leave a comment here and I'll generate a test case, as time allows.

This is meant for "nice to have" tests, and not as a general way to avoid including tests with future PRs

**Comment by skmp** [comment]
From #1752 
- [ ] "This means non-tso code might run but a short while after signal return, even if shared mode has been entered."
- [ ] "When rest of the fragment right after clone, mmap ipc or shmat does some memory operation that needs to be TSO"

**Comment by skmp** [comment]
From SMC work follow ups

#1710
#1698
#1679
#1676

**Comment by skmp** [comment]
Spurious error in armv8.4 runner

```
12/70 Test #11642: pthread_cancel.64.jit.flt ...........................***Failed    0.48 sec
[Info] Migrating to shared memory mode
thread_func(): started; cancellation disabled
main(): sending cancellation request
thread_func(): about to enable cancellation
key_dtor: Thread aborted
main(): thread was canceled
pure virtual method called
terminate called without an active exception
test failed, expected is 0 but got -6
```

These are likely after the guest's main has returned, probably from fex
```
pure virtual method called
terminate called without an active exception
```

run: https://github.com/FEX-Emu/FEX/runs/6842603702?check_suite_focus=true

**Comment by skmp** [comment]
smc-mt-2.x64 intermittent failure

https://github.com/FEX-Emu/FEX/runs/6857393970?check_suite_focus=true
<details>
```
Starting Iteration
Generating code on thread
Waiting for code to be modified
Modifying code from another thread
Waiting for thread to get unblocked
Thread exiting
Iteration should finish now
Iteration done
Starting Iteration
Generating code on thread
Waiting for code to be modified
Modifying code from another thread
Waiting for thread to get unblocked
Thread exiting
Iteration should finish now
Iteration done
Starting Iteration
Generating code on thread
Waiting for code to be modified
Modifying code from another thread
Waiting for thread to get unblocked
Thread exiting
Iteration should finish now
Iteration done
Starting Iteration
Generating code on thread
Waiting for code to be modified
Modifying code from another thread
Waiting for thread to get unblocked
Thread exiting
Iteration should finish now
Iteration done
Starting Iteration
Generating code on thread
Waiting for code to be modified
Modifying code from another thread
Waiting for thread to get unblocked
Thread exiting
Iteration should finish now
Iteration done
Starting Iteration
Generating code on thread
Waiting for code to be modified
Modifying code from another thread
Waiting for thread to get unblocked
Thread exiting
Thread overshoot once, this is non fatal
Iteration should finish now
Iteration done
Starting Iteration
Generating code on thread
Waiting for code to be modified
Modifying code from another thread
Waiting for thread to get unblocked
Thread overshoot once, this is non fatal
Thread exiting
Iteration should finish now
Iteration done
Starting Iteration
Generating code on thread
Waiting for code to be modified
Modifying code from another thread
Waiting for thread to get unblocked
Thread exiting
Iteration should finish now
Iteration done
Starting Iteration
Generating code on thread
Waiting for code to be modified
Modifying code from another thread
Waiting for thread to get unblocked
Thread exiting
Iteration should finish now
Iteration done
Starting Iteration
Generating code on thread
Waiting for code to be modified
Modifying code from another thread
Waiting for thread to get unblocked
Thread exiting
Iteration should finish now
Iteration done
Starting Iteration
Generating code on thread
Waiting for code to be modified
Modifying code from another thread
Waiting for thread to get unblocked
Thread exiting
Iteration should finish now
Iteration done
Starting Iteration
Generating code on thread
Waiting for code to be modified
Modifying code from another thread
Waiting for thread to get unblocked
Thread exiting
Iteration should finish now
Iteration done
Starting Iteration
Generating code on thread
Waiting for code to be modified
Modifying code from another thread
Waiting for thread to get unblocked
Thread exiting
Iteration should finish now
Iteration done
Starting Iteration
Generating code on thread
Waiting for code to be modified
Modifying code from another thread
Waiting for thread to get unblocked
Thread exiting
Iteration should finish now
Iteration done
Starting Iteration
Generating code on thread
Waiting for code to be modified
Modifying code from another thread
Waiting for thread to get unblocked
Thread exiting
Iteration should finish now
Iteration done
Starting Iteration
Generating code on thread
Waiting for code to be modified
Modifying code from another thread
Waiting for thread to get unblocked
Thread exiting
Iteration should finish now
Iteration done
Starting Iteration
Generating code on thread
Waiting for code to be modified
Modifying code from another thread
Waiting for thread to get unblocked
Thread exiting
Iteration should finish now
Iteration done
Starting Iteration
Generating code on thread
Waiting for code to be modified
Modifying code from another thread
Waiting for thread to get unblocked
Thread exiting
Iteration should finish now
Iteration done
Starting Iteration
Generating code on thread
Waiting for code to be modified
Modifying code from another thread
Waiting for thread to get unblocked
Thread exiting
Iteration should finish now
Iteration done
Starting Iteration
Generating code on thread
Waiting for code to be modified
Modifying code from another thread
Waiting for thread to get unblocked
Thread exiting
Iteration should finish now
Iteration done
Starting Iteration
Generating code on thread
Waiting for code to be modified
Modifying code from another thread
Waiting for thread to get unblocked
Thread exiting
Iteration should finish now
Iteration done
Starting Iteration
Generating code on thread
Waiting for code to be modified
Modifying code from another thread
Waiting for thread to get unblocked
Thread exiting
Iteration should finish now
Iteration done
Starting Iteration
Generating code on thread
Waiting for code to be modified
Modifying code from another thread
Waiting for thread to get unblocked
Thread exiting
Iteration should finish now
Iteration done
Starting Iteration
Generating code on thread
Waiting for code to be modified
Modifying code from another thread
Waiting for thread to get unblocked
Thread overshoot once, this is non fatal
Iteration should finish now
Thread exiting
Iteration done
Starting Iteration
Generating code on thread
Waiting for code to be modified
Modifying code from another thread
Waiting for thread to get unblocked
Thread exiting
Thread overshoot once, this is non fatal
Iteration should finish now
Iteration done
Starting Iteration
Generating code on thread
Waiting for code to be modified
Modifying code from another thread
Waiting for thread to get unblocked
Thread overshoot once, this is non fatal
Iteration should finish now
Thread exiting
Iteration done
Starting Iteration
Generating code on thread
Waiting for code to be modified
Modifying code from another thread
Waiting for thread to get unblocked
Thread exiting
Iteration should finish now
Iteration done
Starting Iteration
Generating code on thread
Waiting for code to be modified
Modifying code from another thread
Waiting for thread to get unblocked
Thread exiting
Iteration should finish now
Iteration done
Starting Iteration
Generating code on thread
Waiting for code to be modified
Modifying code from another thread
Waiting for thread to get unblocked
Thread exiting
Iteration should finish now
Iteration done
Starting Iteration
Generating code on thread
Waiting for code to be modified
Modifying code from another thread
Waiting for thread to get unblocked
Thread exiting
Iteration should finish now
Iteration done
Starting Iteration
Generating code on thread
Waiting for code to be modified
Modifying code from another thread
Waiting for thread to get unblocked
Thread exiting
Iteration should finish now
Iteration done
Starting Iteration
Generating code on thread
Waiting for code to be modified
Modifying code from another thread
Waiting for thread to get unblocked
Thread exiting
Iteration should finish now
Iteration done
Starting Iteration
Generating code on thread
Waiting for code to be modified
Modifying code from another thread
Waiting for thread to get unblocked
Thread exiting
Iteration should finish now
Iteration done
Starting Iteration
Generating code on thread
Waiting for code to be modified
Modifying code from another thread
Waiting for thread to get unblocked
Thread overshoot once, this is non fatal
Thread exiting
Iteration should finish now
Iteration done
Starting Iteration
Generating code on thread
Waiting for code to be modified
Modifying code from another thread
Waiting for thread to get unblocked
Thread exiting
Iteration should finish now
Iteration done
Starting Iteration
Generating code on thread
Waiting for code to be modified
Modifying code from another thread
Waiting for thread to get unblocked
Thread exiting
Iteration should finish now
Iteration done
Starting Iteration
Generating code on thread
Waiting for code to be modified
Modifying code from another thread
Waiting for thread to get unblocked
Thread exiting
Iteration should finish now
Iteration done
Starting Iteration
Generating code on thread
Waiting for code to be modified
Modifying code from another thread
Waiting for thread to get unblocked
Thread exiting
Iteration should finish now
Iteration done
Starting Iteration
Generating code on thread
Waiting for code to be modified
Modifying code from another thread
Waiting for thread to get unblocked
Iteration should finish now
Thread exiting
Iteration done
Starting Iteration
Generating code on thread
Waiting for code to be modified
Modifying code from another thread
Waiting for thread to get unblocked
Thread exiting
Iteration should finish now
Iteration done
Starting Iteration
Generating code on thread
Waiting for code to be modified
Modifying code from another thread
Waiting for thread to get unblocked
Thread exiting
Iteration should finish now
Iteration done
Starting Iteration
Generating code on thread
Waiting for code to be modified
Modifying code from another thread
Waiting for thread to get unblocked
Thread exiting
Iteration should finish now
Iteration done
Starting Iteration
Generating code on thread
Waiting for code to be modified
Modifying code from another thread
Waiting for thread to get unblocked
Thread exiting
Iteration should finish now
Iteration done
Starting Iteration
Generating code on thread
Waiting for code to be modified
Modifying code from another thread
Waiting for thread to get unblocked
Thread overshoot once, this is non fatal
Thread exiting
Iteration should finish now
Iteration done
Starting Iteration
Generating code on thread
Waiting for code to be modified
Modifying code from another thread
Waiting for thread to get unblocked
Thread overshoot once, this is non fatal
Thread exiting
Iteration should finish now
Iteration done
Starting Iteration
Generating code on thread
Waiting for code to be modified
Modifying code from another thread
Waiting for thread to get unblocked
Thread overshoot once, this is non fatal
Thread exiting
Iteration should finish now
Iteration done
Starting Iteration
Generating code on thread
Waiting for code to be modified
Modifying code from another thread
Waiting for thread to get unblocked
Thread exiting
Iteration should finish now
Iteration done
Starting Iteration
Generating code on thread
Waiting for code to be modified
Modifying code from another thread
Waiting for thread to get unblocked
Thread exiting
Iteration should finish now
Iteration done
Starting Iteration
Generating code on thread
Waiting for code to be modified
Modifying code from another thread
Waiting for thread to get unblocked
Thread exiting
Thread overshoot once, this is non fatal
Iteration should finish now
Iteration done
Starting Iteration
Generating code on thread
Waiting for code to be modified
Modifying code from another thread
Waiting for thread to get unblocked
Thread overshoot once, this is non fatal
Thread should have been patched to not modify counter here
Thread exiting
test failed, expected is 0 but got 1
```
</details>

**Comment by skmp** [comment]
smc-mt-2.64.host.flt  intermittent failure

https://github.com/FEX-Emu/FEX/runs/7045388463?check_suite_focus=true
<details>
Starting Iteration
Generating code on thread
Waiting for code to be modified
Modifying code from another thread
Waiting for thread to get unblocked
Thread overshoot once, this is non fatal
Iteration should finish now
Thread exiting
Iteration done
Starting Iteration
Generating code on thread
Waiting for code to be modified
Modifying code from another thread
Waiting for thread to get unblocked
Thread exiting
Iteration should finish now
Iteration done
Starting Iteration
Generating code on thread
Waiting for code to be modified
Modifying code from another thread
Waiting for thread to get unblocked
Thread exiting
Iteration should finish now
Iteration done
Starting Iteration
Generating code on thread
Waiting for code to be modified
Modifying code from another thread
Waiting for thread to get unblocked
Thread exiting
Iteration should finish now
Iteration done
Starting Iteration
Generating code on thread
Waiting for code to be modified
Modifying code from another thread
Waiting for thread to get unblocked
Thread exiting
Iteration should finish now
Iteration done
Starting Iteration
Generating code on thread
Waiting for code to be modified
Modifying code from another thread
Waiting for thread to get unblocked
Thread exiting
Iteration should finish now
Iteration done
Starting Iteration
Generating code on thread
Waiting for code to be modified
Modifying code from another thread
Waiting for thread to get unblocked
Thread overshoot once, this is non fatal
Thread exiting
Iteration should finish now
Iteration done
Starting Iteration
Generating code on thread
Waiting for code to be modified
Modifying code from another thread
Waiting for thread to get unblocked
Thread overshoot once, this is non fatal
Thread exiting
Iteration should finish now
Iteration done
Starting Iteration
Generating code on thread
Waiting for code to be modified
Modifying code from another thread
Waiting for thread to get unblocked
Thread overshoot once, this is non fatal
Thread should have been patched to not modify counter here
Thread exiting
test failed, expected is 0 but got 1
</details>

**Comment by skmp** [comment]
https://github.com/FEX-Emu/FEX/runs/8109934359?check_suite_focus=true

```
20/83 Test #11664: pthread_cancel.64.jit.flt ...........................***Failed    0.49 sec
thread_func(): started; cancellation disabled
main(): sending cancellation request
thread_func(): about to enable cancellation
key_dtor: Thread aborted

main(): thread was canceled
pure virtual method called
terminate called without an active exception
test failed, expected is 0 but got -6
```

**Comment by neobrain** [comment]
These notes are probably not any further useful nowadays.
